tests/foreman/cli/test_computeresource_osp.py

```diff
@@ -47,8 +44,15 @@ def cr_cleanup(self, cr_id):
 
     @pytest.mark.upgrade
     @pytest.mark.tier3
-    @pytest.mark.parametrize('id_type', ['id', 'name'])
-    def test_crud_and_duplicate_name(self, request, id_type):
+    @pytest.mark.parametrize(
+        'hostname',
+        [settings.osp.hostname.osp16, settings.osp.hostname.osp17],
+    )
+    @pytest.mark.parametrize(
+        'id_type',
+        ['id', 'name'],
```

Comment:
It tries to search, update, delete CR by Id as well as name

Reply:
I think testing with `id` and `name` is overkill for what the purpose of the test is. Also now that we are parametrizing for `hostname` the test will run 4 times. If you really want to test both `id` and `name` you could change the test easily to do `id` when running `hostname == settings.osp.hostname.osp16` and `name` when `hostname == settings.osp.hostname.osp17`

Reply:
Yes, the test will run 4 times now, but I don't think It will be a good idea to test `id` on one instance and `name` on another. Also running 4 times won't be overkill, the total execution itself takes less than 3 mins. 
For now I'll prefer to keep as it is.

Reply:
I agree with @shubhamsg199 , we should test both CRs for both id and name, and considering a execution time I don't see it as a blocker for merge, still this can be improved later on.